app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,46 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find(params[:id])
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save!
+      redirect_to tasks_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def update
+    @task = Task.find(params[:id])
+    @task.update(task_params)
+    redirect_to tasks_path
+  end
+
+  def destroy
+    @task = Task.find(params[:id])
```

Comment:
I noticed these were repeated 4 times in this controller. I think this would be a great candidate to extract out into a separate method!

Reply:
yes, good catch :) our line of thought was to get the respective parts of tasks and tasklists CRUD actions and views first, and we will refactor from there. Since tasks will be nested in tasklist, was thinking to prepare tasklist instead. Will work on it :)